### PR TITLE
Added Google analytics collection.

### PR DESCRIPTION
**Pull request by mikeller**

### CODE REVIEW

File: src/js/main.js [hunk 1]
```diff
@@ -1,19 +1,67 @@
...
 openNewWindowsInExternalBrowser();
 
-//Asynchronous configuration to be done.
-//When finish the startProcess() function must be called 
+// Asynchronous configuration to be done.
+// When finish the startProcess() function must be called
 $(document).ready(function () {
     i18n.init(function() {
-        startProcess();
+        setupAnalytics();
```
Comment (McGiverGim) [inline src/js/main.js]: You changed the method but not the comment about the method.

File: src/js/main.js [hunk 2]
```diff
@@ -1,19 +1,67 @@
...
+        if (result.userId) {
+            userId = result.userId;
+        } else {
+            var uid = new ShortUniqueId();
+            userId = uid.randomUUID(13);
+
+            chrome.storage.local.set({ 'userId': userId });
+        }
+
+        var optOut = !!result.analyticsOptOut
```
Comment (McGiverGim) [inline src/js/main.js]: Add termination line ;

**Comment by mikeller** [comment]
@McGiverGim: Just to keep you from getting a heart attack :wink: : The policy should **not** be translated, it was auto-generated with all the required legalese in place, and we need it because having a privacy policy is one of the requirements of Google for being allowed to use Google Analytics.

**Comment by mikeller** [comment]
Ok, this is done now.

File: src/js/Analytics.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    this.resetFirmwareData();
+};
+
+Analytics.prototype.setDimension = function (dimension, value) {
+    var dimensionName = 'dimension' + dimension;
+    this._analytics.custom(dimensionName, value);
+}
+
+Analytics.prototype.sendEvent = function (category, action, options) {
+//    options.eventLabel = options.eventLabel || this._flightControllerData[this.DATA.MCU_ID];
```
Comment (McGiverGim) [inline src/js/Analytics.js]: Why is this line commented? Best add a comment explaining it or remove the line.

Comment (mikeller) [reply src/js/Analytics.js]: Hell yeah, looks like our best practices work! This a leftover from debugging, and should have been removed. Will change, thanks for the review.

**Pull request merged**